crypto/x509/v3_purp.c

```diff
@@ -644,19 +620,7 @@ static int purpose_smime(const X509 *x, int ca)
         ca_ret = check_ca(x);
         if (!ca_ret)
             return 0;
-        /* check nsCertType if present */
-        if (ca_ret != 5 || x->ex_nscert & NS_SMIME_CA)
-            return ca_ret;
-        else
-            return 0;
-    }
-    if (x->ex_flags & EXFLAG_NSCERT) {
-        if (x->ex_nscert & NS_SMIME)
-            return 1;
-        /* Workaround for some buggy certificates */
-        if (x->ex_nscert & NS_SSL_CLIENT)
-            return 2;
-        return 0;
+        return ca_ret;
```

Comment:
The two lines above this are redundant with this line.
However, the original code would return 0 if ca_ret = 5.